examples/lsc/mag240m/gnn.py

```diff
@@ -13,31 +13,25 @@
 from torch.nn import ModuleList, Sequential, Linear, BatchNorm1d, ReLU, Dropout
 from torch.optim.lr_scheduler import StepLR
 
-from pytorch_lightning.metrics import Accuracy
+from torchmetrics import Accuracy
 from pytorch_lightning.callbacks import ModelCheckpoint
 from pytorch_lightning import (LightningDataModule, LightningModule, Trainer,
                                seed_everything)
 
 from torch_sparse import SparseTensor
-from torch_geometric.nn import SAGEConv, GATConv
+from torch_geometric.nn import SAGEConv, GATConv, to_hetero
 from torch_geometric.data import NeighborSampler
 
 from ogb.lsc import MAG240MDataset, MAG240MEvaluator
 from root import ROOT
+from torch_geometric.loader.neighbor_loader import NeighborLoader
+from torch_geometric.typing import Adj
+import torch_geometric.transforms as T
+from torch_geometric.typing import EdgeType, NodeType
+from typing import Dict, Tuple
+from torch_geometric.data import Batch
 
-
-class Batch(NamedTuple):
-    x: Tensor
-    y: Tensor
-    adjs_t: List[SparseTensor]
-
-    def to(self, *args, **kwargs):
-        return Batch(
-            x=self.x.to(*args, **kwargs),
-            y=self.y.to(*args, **kwargs),
-            adjs_t=[adj_t.to(*args, **kwargs) for adj_t in self.adjs_t],
-        )
-
+device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
 
 class MAG240M(LightningDataModule):
```

Comment:
We could try to make use of `torch_geometric.data.LightningNodeDataset` for this. This would simplify the construction of neighbor loaders.

Reply:
Sorry. There is no `LightningNodeDataset` in pyg.

Reply:
You mean `LightningNodeData`? Will try this.

Reply:
I have updated the code using `LightningNodeData`, but it still get the RuntimeError `Node conv1__paper1 target conv1.author__writes__paper references nonexistent attribute author__writes__paper of conv1`.